packages/codemod/src/transforms/react-default-props-migration.ts

```diff
@@ -0,0 +1,426 @@
+          );
+        }
+
+        // 3. Next we update the component function signature
+        // We first look for classic function declarations
+        //   function MyComnponent(props) { ... }
+        const functionComponentDeclaration = root.find(j.FunctionDeclaration, {
+          id: { name: componentName },
+        });
+        if (functionComponentDeclaration.length === 1) {
+          functionComponentDeclaration.nodes()[0].params[0] =
+            transformComponentFunctionSignature({
+              functionPropsParam:
+                functionComponentDeclaration.nodes()[0].params[0],
+              defaultPropsMap,
+              componentName,
+              j,
+            });
+
+          // Get the function body scope so we only do the replacement
+          // within the component function we're currently processing
+          functionScope = j(functionComponentDeclaration.nodes()[0].body);
+        } else {
+          // If we don't find a function declaration, we look for arrow function declarations
+          //   const MyComnponent = (props) => { ... }
+          //   const MyComnponent = ({ prop1, ...props }) => { ... }
+          const variableComponentDeclaration = root.find(
+            j.VariableDeclaration,
+            {
+              declarations: [
+                {
+                  id: { name: componentName },
+                },
+              ],
+            }
+          );
+          if (variableComponentDeclaration.length === 1) {
+            const functionFirstParamNode =
+              variableComponentDeclaration.nodes()[0].declarations[0];
+
+            if (
+              functionFirstParamNode.type === 'VariableDeclarator' &&
+              functionFirstParamNode.init?.type === 'ArrowFunctionExpression'
+            ) {
+              functionFirstParamNode.init.params[0] =
+                transformComponentFunctionSignature({
+                  functionPropsParam: functionFirstParamNode.init.params[0],
+                  defaultPropsMap,
+                  componentName,
+                  j,
+                });
+
+              // Get the function body scope so we only do the replacement
+              // within the component function we're currently processing
+              functionScope = j(functionFirstParamNode.init.body);
+            } else {
+              console.warn(
+                `[WARNING]: Could parse component function first parameter "${componentName}"`
+              );
+            }
+          }
+        }
+
+        // 4. Refactor the usages of the default props in the body of the component
+        replacePropsUsage({
+          j,
+          defaultPropsKeys: Object.keys(defaultPropsMap),
+          scope: functionScope!,
+        });
+
+        // 5. Update the component TS type definition so we make sure the default props are optional
+        updateComponentTypes({
+          j,
+          root,
+          typeName: `${componentName}Props`,
```

Comment:
This is also something I noticed, if we build the typeName from this, sometimes, there are cases where the typeName is inconsistent and we end up having files with typeNames as just `${componentName}` without the Props suffix. 
This also would have to be updated manually for cases like these.

Reply:
Yes, this relies on us using a consistent naming.
I think I can improve it further as a follow-up.

Reply:
We can remove the suffix `Props` from the existing name to make sure we can append it. If the component name doesn't end up with `Props`, the `replace` doesn't do anything and we can still append the suffix correctly.

```suggestion
          typeName: `${componentName.replace(/(Props)$/), ''}Props`,
```

Reply:
To be honest, I run this codemod on the 100+ component in the `ui-kit` repository and didn't find this problem so I would prefer to actually have a use case for changing it.

Reply:
I've just realized I was wrong. I had to adjust some things manually in `ui-kit` but didn't think it was because of this.
However, I would like to do it as a follow-up in order to not expand the scope of this PR.

Reply:
> I would like to do it as a follow-up in order to not expand the scope of this PR

Do you mean adding `.replace(/(Props)$/, '')`? Or are you referring to other things?

Reply:
I had a more elaborated idea in mind but I already implemented while I was waiting for the review.

Please take a look at this commit: 9af02f73a9e11464c810ba5834ff9c7409ea18f3